x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Expression.java

```diff
@@ -207,4 +219,38 @@ public String toString() {
     public String propertiesToString(boolean skipIfChild) {
         return super.propertiesToString(false);
     }
+
+    protected static boolean isAttributePushable(
+        Expression expression,
+        Expression operation,
+        Predicate<FieldAttribute> hasIdenticalDelegate
+    ) {
+        if (isPushableFieldAttribute(expression, hasIdenticalDelegate)) {
+            return true;
+        }
+        if (expression instanceof MetadataAttribute ma && ma.searchable()) {
+            return operation == null
+                // no range or regex queries supported with metadata fields
+                || operation instanceof Equals
+                || operation instanceof NotEquals
+                || operation instanceof WildcardLike;
```

Comment:
I wonder if instead of checking the instance type here, we should override this on the relevant operators?  That would also work around the fact that we can't see the right classes from here.

Reply:
I just did that :)